cmd/ci-secret-generator/main.go

```diff
@@ -120,17 +120,34 @@ func (o *options) validateConfig() error {
 }
 
 func executeCommand(command string) ([]byte, error) {
-	out, err := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", command).CombinedOutput()
-	if err != nil {
-		return nil, fmt.Errorf("%s : %w", string(out), err)
+	cmd := exec.Command("bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c", command)
+	var outBuf, errBuf bytes.Buffer
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	if err := cmd.Run(); err != nil {
+		// The command completed with non zero exit code, standard streams *should* be available.
+		if _, ok := err.(*exec.ExitError); ok {
+			errStr := errBuf.String()
+			return nil, fmt.Errorf("failed to run the command %q: %w\nerror output: %q",
+				command, err, errStr)
+		}
+		// At this point we can't easily tell why the command failed,
+		// there is no guarantee neither stdout nor stderr are valid.
+		return nil, fmt.Errorf("failed to run the command %q: %w", command, err)
 	}
-	if len(out) == 0 || len(bytes.TrimSpace(out)) == 0 {
+
+	if len(errBuf.Bytes()) != 0 {
+		return nil, fmt.Errorf("command %q has error output", command)
+	}
+
+	stdout := outBuf.Bytes()
+	if len(stdout) == 0 || len(bytes.TrimSpace(stdout)) == 0 {
 		return nil, fmt.Errorf("command %q returned no output", command)
 	}
-	if string(bytes.TrimSpace(out)) == "null" {
+	if string(bytes.TrimSpace(stdout)) == "null" {
```

Comment:
Not related to the PR.
I wonder why "null" leads to an error. Unlikely to cause any damage though.

Reply:
Really have no idea why, I need to dig deeper eventually.

Reply:
My guess is it's for `jq`, we have some instances in the configuration file.

(why the check is in the program, not the script, I do not understand)

Reply:
```
$ jq '.test' <<< '{}'
null
```

Reply:
We should consider opening a PR to refactor that.